File: src/main/java/com/kdopamine/app/domain/sentence/dto/response/SentenceSubmitRes.java
package com.kdopamine.app.domain.sentence.dto.response;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.kdopamine.app.domain.sentence.entity.Sentence;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class SentenceSubmitRes {
    @JsonProperty("isCorrect")
    private boolean isCorrect;
    private String userAnswer;
    private String correctAnswer;
    private String voiceFileUrl;

    public static SentenceSubmitRes of(boolean isCorrect, String userAnswer, Sentence sentence){

        String cleanAnswer = sentence.getKorean().replaceAll("\\{([^:}]+):([^}]+)\\}", "$1");

        return SentenceSubmitRes.builder()
                .isCorrect(isCorrect)
                .userAnswer(userAnswer)
                .correctAnswer(cleanAnswer)
                .voiceFileUrl(sentence.getVoiceFileUrl())
                .build();
    }
}

File: k_lingo_front/lib/models/study/sentence_submit_res.dart
class SentenceSubmitRes {
  final bool isCorrect;
  final String userAnswer;
  final String correctAnswer;
  final String? voiceFileUrl;

  SentenceSubmitRes({
    required this.isCorrect,
    required this.userAnswer,
    required this.correctAnswer,
    this.voiceFileUrl,
  });

  factory SentenceSubmitRes.fromJson(Map<String, dynamic> json) {
    return SentenceSubmitRes(
      isCorrect: json['isCorrect'] ?? false,
      userAnswer: json['userAnswer'] ?? '',
      correctAnswer: json['correctAnswer'] ?? '',
      voiceFileUrl: json['voiceFileUrl'],
    );
  }
}